packages/stream_chat/lib/src/api/channel.dart

```diff
@@ -454,20 +451,19 @@ class Channel {
   }
 
   /// Pins provided message
-  Future<UpdateMessageResponse> pinMessage(
+  Future<UpdateMessageResponse?> pinMessage(
     Message message,
     Object timeoutOrExpirationDate,
   ) {
     assert(() {
       if (timeoutOrExpirationDate is! DateTime &&
-          timeoutOrExpirationDate is! num &&
-          timeoutOrExpirationDate != null) {
```

Comment:
`timeoutOrExpirationData` should marked nullable and this line should be added back because we are  allowing it to be set to null in order to totally remove the expiration date.